cunumeric/array.py

```diff
@@ -3046,13 +3155,23 @@ def setflags(self, write=None, align=None, uic=None):
         Multiple GPUs, Multiple CPUs
 
         """
-        self.__array__().setflags(write=write, align=align, uic=uic)
+        # Be a bit more careful here, and only pass params that are explicitly
+        # set by the caller. The numpy interface specifies only bool values,
+        # despite its None defaults.
+        kws = {}
+        if write is not None:
+            kws["write"] = write
+        if align is not None:
+            kws["align"] = align
+        if uic is not None:
+            kws["uic"] = uic
+        self.__array__().setflags(**kws)
```

Comment:
Being extra careful here, plus this is easier to type. TLDR this only passes values to `ndarray.setflags` if they were explicity passed in to us.